lib/fswatcher/fswatcher.go

```diff
@@ -145,31 +166,77 @@ func (watcher *FsWatcher) speedUpNotifyTimer() {
 }
 
 func (watcher *FsWatcher) slowDownNotifyTimer() {
-	if watcher.notifyDelay != slowNotifyDelay {
-		watcher.notifyDelay = slowNotifyDelay
+	if watcher.notifyDelay != watcher.slowNotifyDelay {
+		watcher.notifyDelay = watcher.slowNotifyDelay
 		watcher.debugf("Slowing down notifyTimer to %#v\n", watcher.notifyDelay)
 		watcher.notifyTimerNeedsReset = true
 	}
 }
 
-func (watcher *FsWatcher) storeFsEvent(event *FsEvent) {
-	if watcher.pathInProgress(event.path) {
-		watcher.debugf("Skipping notification for finished path: %s\n",
-			event.path)
+func (watcher *FsWatcher) aggregateEvent(path string, eventTime time.Time) bool {
+	if filepath.Clean(path) == "." {
+		watcher.fsEvents = make(FsEventsBatch)
+		watcher.fsEvents["."] = &FsEvent{".", eventTime}
+		return true
+	}
+	dirPath := filepath.Dir(path)
+	localMaxFilesPerDir := maxFilesPerDir
+	if dirPath == "." {
+		localMaxFilesPerDir = maxFiles
+	}
+	if _, ok := watcher.fsEvents[dirPath]; ok {
+		watcher.debugf("Aggregating: Parent directory already tracked: %s", path)
+		delete(watcher.fsEvents, path)
+		return false
+	} else if dir, ok := watcher.trackedDirs[dirPath]; ok && dir.count == localMaxFilesPerDir {
```

Comment:
Because you're returning in the if, life would be easier if the else clause did not exist, as in

if x {
  return 1
}
if y { 
  return 2
}